fineract-provider/src/main/java/org/apache/fineract/portfolio/collateralmanagement/service/LoanCollateralAssembler.java

```diff
@@ -0,0 +1,132 @@
+
+            for (int i = 0; i < collaterals.size(); i++) {
+                final JsonObject collateralItemElement = collaterals.get(i).getAsJsonObject();
+                final Long id = this.fromApiJsonHelper.extractLongNamed("id", collateralItemElement);
+                final Long collateralId = this.fromApiJsonHelper.extractLongNamed("clientCollateralId", collateralItemElement);
+                final ClientCollateralManagement clientCollateral = this.clientCollateralManagementRepositoryWrapper
+                        .getCollateral(collateralId);
+                final BigDecimal quantity = this.fromApiJsonHelper.extractBigDecimalNamed("quantity", collateralItemElement, locale);
+                BigDecimal updatedClientQuantity = null;
+
+                if (id == null) {
+                    updatedClientQuantity = clientCollateral.getQuantity().subtract(quantity);
+                    if (BigDecimal.ZERO.compareTo(updatedClientQuantity) > 0) {
+                        throw new InvalidAmountOfCollateralQuantity(quantity);
+                    }
+                    clientCollateral.updateQuantity(updatedClientQuantity);
+                    // this.clientCollateralManagementRepositoryWrapper.saveAndFlush(clientCollateral);
+                    collateralItems.add(LoanCollateralManagement.from(clientCollateral, quantity));
+                } else {
+                    LoanCollateralManagement loanCollateralManagement = this.loanCollateralRepository.findById(id)
+                            .orElseThrow(() -> new LoanCollateralManagementNotFoundException(id));
+
+                    if (loanCollateralManagement.getQuantity().compareTo(quantity) != 0) {
+                        updatedClientQuantity = clientCollateral.getQuantity().add(loanCollateralManagement.getQuantity())
+                                .subtract(quantity);
+                        if (BigDecimal.ZERO.compareTo(updatedClientQuantity) > 0) {
+                            throw new InvalidAmountOfCollateralQuantity(quantity);
+                        }
+                    } else {
+                        updatedClientQuantity = quantity;
+                    }
+
+                    // loanCollateralManagement.setQuantity(quantity);
+                    clientCollateral.updateQuantity(updatedClientQuantity);
+                    // loanCollateralManagement.setClientCollateralManagement(clientCollateral);
+                    // this.clientCollateralManagementRepositoryWrapper.saveAndFlush(clientCollateral);
+                    collateralItems
+                            .add(LoanCollateralManagement.fromExisting(clientCollateral, quantity, loanCollateralManagement.getLoanData(),
+                                    loanCollateralManagement.getLoanTransaction(), loanCollateralManagement.getId()));
+                }
+            }
+        }
+
+        // if (element.isJsonObject()) {
```

Comment:
Please check these commented-out code blocks if they are still relevant. Please add more details if they will be needed otherwise remove them.

Reply:
Fixed. Removed unwanted comment